python/pyspark/sql/profiler.py

```diff
@@ -224,6 +224,54 @@ def dump(id: int) -> None:
             for id in sorted(code_map.keys()):
                 dump(id)
 
+    def clear_perf_profiles(self, id: Optional[int] = None) -> None:
+        """
+        Clear the perf profile results.
+
+        .. versionadded:: 4.0.0
+
+        Parameters
+        ----------
+        id : int, optional
+            The UDF ID whose profiling results should be cleared.
+            If not specified, all the results will be cleared.
+        """
+        ids_to_remove = [
+            result_id
+            for result_id, (perf, _, *_) in self._profile_results.items()
+            if perf is not None
+        ]
+        with self._lock:
+            if id is not None:
+                if id in ids_to_remove:
+                    self._profile_results.pop(id, None)
```

Comment:
Seems to be removing `memory` as well?

Reply:
On Jupyter:

```py
from pyspark.sql.functions import pandas_udf
df = spark.range(3)

@pandas_udf("long")
def add1(x):
    return x + 1

added = df.select(add1("id"))

spark.conf.set("spark.sql.pyspark.udf.profiler", "perf")
added.show()

spark.conf.set("spark.sql.pyspark.udf.profiler", "memory")
added.show()

spark.profile.show()
...

spark.profile.clear(type="memory")

spark.profile.show()  # should still show the perf results?
```

Reply:
Good catch! Thanks for the example. I adjusted the code and added tests.